README.md

```diff
@@ -444,68 +417,60 @@ transcoder.extend(
-
-We might like for an encoder to always output the shortest possible payload for a given value, whichever form that may be. The most accurate way to determine this is to encode the input value to both forms and return the shorter one, but this is rather laborious. Much simpler determinations can be correct in almost all cases, and an optimised form payload will almost never be more than a few bytes larger than the equivalent simple form payload. For these reasons, the spec writers consider it reasonable to use the optimised form whenever the payload contains a map or repeated string, and to use the simple form otherwise. This rule of thumb allows for some uncommon cases where the optimised form is used when the simple form would be smaller, but it allows for a much easier determination, will never miss an optimisation, and still avoids using the optimised form when it just adds an obviously-useless 3-byte prefix of `opts 0 array5` to the equivalent simple form.
-
-### Repeated String Optimisation
-
-The string lookup table stores strings which occur multiple times, allowing them to be referenced by the `strref` type tag so repeated instances each take constant space regardless of the size of the string.
-
-Specifically, the string lookup table has a maximum size of 255, so as to guarantee that a string reference requires only two bytes.
-
-### Repeated Keyset Optimisation
-
-TODO: explain
-
-### Precautions
-
-As with any compression-like scheme, highly-compressed inputs which result in a much larger decoded value are a concern.
-
-For SuperPack payloads crafted for maximum expansion, the following relationships between input size and expanded size hold:
+```js
+let symbols = [];
+let decodedSymbols = [];
+
+// preserves ECMAScript Symbol identity and description
+let symbolExtension = {
+  detector: x => typeof x === 'symbol',
+  serialiser(s) {
+    let i = symbols.indexOf(s);
+    if (i >= 0) return i;
+    return symbols.push(s) - 1;
+  },
+  deserialiser(n, memo) {
+    if (decodedSymbols[n] == null) {
+      let s = Symbol(memo[n]);
+      decodedSymbols[n] = s;
+      return s;
+    } else {
+      return decodedSymbols[n];
+    }
+  },
+  memo: () => symbols.map(sym => String(sym).slice(7, -1)),
+};
+
+let encoded = encode(data, { extensions: { 0: symbolExtension } });
+```
 
-| Input Size | Expanded Size |
-| ---------- | ------------- |
-| 1kB        | 125kB         |
-| 2kB        | 500kB         |
-| 2.83kB     | 1mB           |
-| 4kB        | 2mB           |
-| 40kB       | 200mB         |
-| 400kB      | 20gB          |
+Memos are only prepended for extensions that define the optional `memo`
+function. Memos are encoded and prepended to the SuperPack payload in
+descending extension point order. During the encoding of a memo, extensions
+that define the optional `memo` function must not be applied if their extension
+point is greater than that of the extension whose memo is being encoded.
```

Comment:
this section seems a bit underspecified. e.g. what is a "memo" exactly? What does it mean to "encode and prepend" a memo?